pkg/network/transports.go

```diff
@@ -39,6 +43,48 @@ func newAutoTransport(v1 http.RoundTripper, v2 http.RoundTripper) http.RoundTrip
 	})
 }
 
+const (
+	initialTO = float64(50 * time.Millisecond)
+	sleepTO   = 30 * time.Millisecond
+	factor    = 1.4
+	numSteps  = 10
+)
+
+var errDialTimeout = errors.New("timed out dialing")
+
+// dialWithBackOff executes `net.Dialer.DialContext()` with exponentially increasing
+// dial timeouts. In addition it sleeps with random jitter between tries.
+func dialWithBackOff(ctx context.Context, network, address string) (net.Conn, error) {
+	return dialBackOffHelper(ctx, network, address, numSteps, initialTO, sleepTO)
+}
+
+func dialBackOffHelper(ctx context.Context, network, address string, steps int, initial float64, sleep time.Duration) (net.Conn, error) {
+	to := initial
+	dialer := &net.Dialer{
+		Timeout:   time.Duration(to),
+		KeepAlive: 5 * time.Second,
+		DualStack: true,
+	}
+	// TODO(vagababov): use backoff.Step when we use moden k8s client.
+	for i := 0; i < steps; i++ {
+		c, err := dialer.DialContext(ctx, network, address)
+		if err != nil {
```

Comment:
Consider unindenting:
```
if c, err := ...; err == nil {
  return c, err
}
// Handle retries
```

Reply:
ah, you'd still have it with `else` due to the type cast.  I saw the redefinition and assumed `err` was unused.